Typescript types for StrictHTMLFormElements are broken in 0.0.6

### Describe the issue

While trying to bump RSD to latest in our fluent repo (https://github.com/microsoft/fluentui/tree/xplat/apps/xplat-v9) I hit the current set of errors during the build step:

<details>
<summary>Error log</summary>





```
node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:39:3 - error TS1131: Property or signature expected.

39   @@iterator(): Iterator<Elem>;
     ~

node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:39:15 - error TS1109: Expression expected.

39   @@iterator(): Iterator<Elem>;


node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:40:3 - error TS1128: Declaration or statement expected.

40   readonly length: number;
     ~~~~~~~~

node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:41:18 - error TS1005: ']' expected.

41   readonly [index: number | string]: Elem;
                    ~

node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:41:35 - error TS1005: ';' expected.

41   readonly [index: number | string]: Elem;
                                     ~

node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:41:36 - error TS1128: Declaration or statement expected.

41   readonly [index: number | string]: Elem;
                                      ~

node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:42:1 - error TS1128: Declaration or statement expected.

42 }
   ~


Found 7 errors in the same file, starting at: node_modules/react-strict-dom/dist/types/StrictHTMLFormElements.d.ts:39
```

</details>

After some digging, it seems that this is caused by this new `StrictHTMLCollection` interface added a few days ago in this commit by @nmn https://github.com/facebook/react-strict-dom/commit/2c06e8437b8c367482a72cb11a16d89c4e065f11

What happens is that the Typescript types generated for it are like so:
```ts
export interface StrictHTMLCollection<Elem extends StrictHTMLElement> {
  @@iterator(): Iterator<Elem>;
  readonly length: number;
  readonly [index: number | string]: Elem;
}
```
In `StrictHTMLFormElements.d.ts` - which I can't link because it looks like the TS types are only generated at bundling time for new npm releases(?)

And that `@@iterator(): Iterator<Elem>;` line is not valid TS syntax. It should be `[Symbol.iterator](): IterableIterator<Elem>;`. So probably it's more of a root problem on `flow-api-translator` but until this type is fixed in RSD we can't bump to 0.0.6.

### Expected behavior

Types are correct and tsc builds successfully.

### Steps to reproduce

1. create a new `react-native` project (`npx react-native@latest init RSDBaseProject`)
2. add `import type {StrictHTMLCollection} from 'react-strict-dom/dist/types/StrictHTMLFormElements';` around L20 or so in `App.tsx`
3. `yarn tsc`

### Test case

_No response_

### Additional comments

_No response_

--- a/packages/react-strict-dom/src/types/StrictHTMLCollection.d.ts
+++ b/packages/react-strict-dom/src/types/StrictHTMLCollection.d.ts
@@ -1,0 +1,16 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type { StrictHTMLElement } from './StrictHTMLElement';
+
+// This interface is needed because flow-api-translator can't translate
+// the Flow type correctly.
+export interface StrictHTMLCollection<out Elem extends StrictHTMLElement> {
+  [Symbol.iterator](): IterableIterator<Elem>;
+  readonly length: number;
+  readonly [index: number]: Elem;
+}

--- a/packages/react-strict-dom/src/types/StrictHTMLCollection.js
+++ b/packages/react-strict-dom/src/types/StrictHTMLCollection.js
@@ -1,0 +1,16 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import type { StrictHTMLElement } from './StrictHTMLElement';
+
+export interface StrictHTMLCollection<+Elem: StrictHTMLElement> {
+  @@iterator(): Iterator<Elem>;
+  +length: number;
+  +[index: number | string]: Elem;
+}

--- a/packages/react-strict-dom/src/types/StrictHTMLFormElements.js
+++ b/packages/react-strict-dom/src/types/StrictHTMLFormElements.js
@@ -7,6 +7,7 @@
  * @flow strict
  */
 
+import type { StrictHTMLCollection } from './StrictHTMLCollection';
 import type { StrictHTMLElement } from './StrictHTMLElement';
 
 type SelectionDirection = 'backward' | 'forward' | 'none';
@@ -41,12 +42,6 @@ export interface StrictHTMLOptionElement extends StrictHTMLElement {
   +selected: boolean;
   +text: string;
   value: string;
-}
-
-export interface StrictHTMLCollection<+Elem: StrictHTMLElement> {
-  @@iterator(): Iterator<Elem>;
-  +length: number;
-  +[index: number | string]: Elem;
 }
 
 export interface StrictHTMLSelectElement extends StrictHTMLElement {

--- a/packages/react-strict-dom/src/types/flowtests/is-subtype-of-strict-flowtest.js.flow
+++ b/packages/react-strict-dom/src/types/flowtests/is-subtype-of-strict-flowtest.js.flow
@@ -7,24 +7,23 @@
  * @flow strict
  */
 
-import type { StrictAnimation } from "../StrictAnimation";
-import type { StrictElement } from "../StrictElement";
-import type { StrictEventTarget } from "../StrictEventTarget";
-import type { StrictHTMLElement } from "../StrictHTMLElement";
-import type {
-  StrictHTMLCollection,
-  StrictHTMLInputElement,
-  StrictHTMLOptionElement,
-  StrictHTMLSelectElement,
-  StrictHTMLTextAreaElement,
-} from "../StrictHTMLFormElements";
-import type { StrictHTMLImageElement } from "../StrictHTMLImageElement";
-import type { StrictNode } from "../StrictNode";
+import type { StrictAnimation } from '../StrictAnimation';
+import type { StrictElement } from '../StrictElement';
+import type { StrictEventTarget } from '../StrictEventTarget';
 
+import type { StrictHTMLCollection } from '../StrictHTMLCollection';
+import type { StrictHTMLElement } from '../StrictHTMLElement';
+import type { StrictHTMLImageElement } from '../StrictHTMLImageElement';
+import type { StrictHTMLInputElement } from '../StrictHTMLFormElements';
+import type { StrictHTMLOptionElement } from '../StrictHTMLFormElements';
+import type { StrictHTMLSelectElement } from '../StrictHTMLFormElements';
+import type { StrictHTMLTextAreaElement } from '../StrictHTMLFormElements';
+
+import type { StrictNode } from '../StrictNode';
 
 // EventTarget
 declare var eventTarget: EventTarget;
-(eventTarget as StrictEventTarget);
+eventTarget as StrictEventTarget;
 
 // Animation extends EventTarget
 // TODO: Uncomment this after D48380371 is in a Flow release
@@ -33,36 +32,36 @@ declare var eventTarget: EventTarget;
 
 // Node extends EventTarget
 declare var node: Node;
-(node as StrictNode);
+node as StrictNode;
 
 // Element extends Node
 declare var element: Element;
-(element as StrictElement);
+element as StrictElement;
 
 declare var htmlCollection: HTMLCollection<HTMLElement>;
-(htmlCollection as StrictHTMLCollection<HTMLElement>);
-(htmlCollection as StrictHTMLCollection<StrictHTMLElement>);
+htmlCollection as StrictHTMLCollection<HTMLElement>;
+htmlCollection as StrictHTMLCollection<StrictHTMLElement>;
 
 // HTMLElement extends Element
 declare var htmlElement: HTMLElement;
-(htmlElement as StrictHTMLElement);
-
-// HTMLInputElement extends HTMLElement
-declare var htmlInputElement: HTMLInputElement;
-(htmlInputElement as StrictHTMLInputElement);
-
-// HTMLOptionElement extends HTMLElement
-declare var htmlOptionElement: HTMLOptionElement;
-(htmlOptionElement as StrictHTMLOptionElement);
-
-// HTMLSelectElement extends HTMLElement
-declare var htmlSelectElement: HTMLSelectElement;
-(htmlSelectElement as StrictHTMLSelectElement);
-
-// HTMLTextAreaElement extends HTMLElement
-declare var htmlTextAreaElement: HTMLTextAreaElement;
-(htmlTextAreaElement as StrictHTMLTextAreaElement);
+htmlElement as StrictHTMLElement;
 
 // HTMLImageElement extends HTMLElement
 declare var htmlImageElement: HTMLImageElement;
-(htmlImageElement as StrictHTMLImageElement);
+htmlImageElement as StrictHTMLImageElement;
+
+// HTMLInputElement extends HTMLElement
+declare var htmlInputElement: HTMLInputElement;
+htmlInputElement as StrictHTMLInputElement;
+
+// HTMLOptionElement extends HTMLElement
+declare var htmlOptionElement: HTMLOptionElement;
+htmlOptionElement as StrictHTMLOptionElement;
+
+// HTMLSelectElement extends HTMLElement
+declare var htmlSelectElement: HTMLSelectElement;
+htmlSelectElement as StrictHTMLSelectElement;
+
+// HTMLTextAreaElement extends HTMLElement
+declare var htmlTextAreaElement: HTMLTextAreaElement;
+htmlTextAreaElement as StrictHTMLTextAreaElement;

--- a/packages/scripts/gen-types.js
+++ b/packages/scripts/gen-types.js
@@ -121,7 +121,11 @@ function preprocessFileContents(inputCode) {
 }
 
 function postProcessTSOutput(outputCode) {
-  const result = outputCode.replace(/<>/g, '');
+  // `flow-api-translator` doesn't translate this Flow syntax correctly
+  const result = outputCode
+    .replace(/@@iterator\(\):/, '[Symbol.iterator]():')
+    .replace(/\bIterator</, 'IterableIterator<')
+    .replace(/<>/g, '');
 
   return result;
 }

